script/deploy/facets/UpdateWhitelistManagerFacet.s.sol

```diff
@@ -44,80 +47,994 @@ contract DeployScript is UpdateScriptBase {
+            }
+        }
+        return merged;
+    }
+
+    /// @notice Helper function to check if a `bytes4` value exists in an array.
+    /// @param arr The array to search.
+    /// @param val The value to find.
+    /// @return true if `val` is in `arr`, false otherwise.
+    function _containsBytes4(
+        bytes4[] memory arr,
+        bytes4 val
+    ) internal pure returns (bool) {
+        for (uint256 i = 0; i < arr.length; i++) {
+            if (arr[i] == val) return true;
+        }
+        return false;
+    }
+
+    /// @notice Parses the ".DEXS" section of the whitelist JSON.
+    /// @dev Uses a two-pass strategy: first count, then allocate, then populate.
+    /// @param whitelistJson The raw JSON string.
+    /// @return contracts Array of addresses from the DEXS section.
+    /// @return selectors Array of selector arrays for each address.
+    function _parseDexsSection(
+        string memory whitelistJson
+    )
+        internal
+        returns (address[] memory contracts, bytes4[][] memory selectors)
+    {
+        // 1. Count total contracts for this network to pre-allocate arrays
+        uint256 totalContracts = _countDexsContracts(whitelistJson);
+
+        // 2. Allocate arrays
+        contracts = new address[](totalContracts);
+        selectors = new bytes4[][](totalContracts);
+
+        // 3. Populate arrays with data
+        uint256 contractIndex = 0;
+        contractIndex = _populateDexsContracts(
+            whitelistJson,
+            contracts,
+            selectors,
+            contractIndex
+        );
+    }
+
+    /// @notice Parses the ".PERIPHERY" section of the whitelist JSON.
+    /// @dev Uses a two-pass strategy: first count, then allocate, then populate.
+    /// @param whitelistJson The raw JSON string.
+    /// @return contracts Array of addresses from the PERIPHERY section.
+    /// @return selectors Array of selector arrays for each address.
+    function _parsePeripherySection(
+        string memory whitelistJson
+    )
+        internal
+        returns (address[] memory contracts, bytes4[][] memory selectors)
+    {
+        // Check if PERIPHERY section exists for current network
+        string memory peripheryKey = string.concat(".PERIPHERY.", network);
+        try vm.parseJson(whitelistJson, peripheryKey) returns (bytes memory) {
+            // 1. Count contracts
+            uint256 totalContracts = _countPeripheryContracts(whitelistJson);
+
+            // 2. Allocate arrays
+            contracts = new address[](totalContracts);
+            selectors = new bytes4[][](totalContracts);
+
+            // 3. Populate arrays
+            uint256 contractIndex = 0;
+            contractIndex = _populatePeripheryContracts(
+                whitelistJson,
+                contracts,
+                selectors,
+                contractIndex
+            );
+        } catch {
+            // Network not found, throw error
+            revert(
```

Comment:
so if for one network we do not have a periphery section then the whole update script fails?
I guess we will have one for each network anyway (except `localanvil`) but I wonder if reverting is the correct strategy here. And did you make sure that we catch the localanvil case?

Actually we could also remove the localanvil I guess...

Just checking to make sure we dont have a non-executable script but I guess you tested it already and it works anyway?